pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -367,6 +367,13 @@ func (asw *actualStateOfWorld) addVolume(
 			devicePath:         devicePath,
 		}
 		asw.attachedVolumes[volumeName] = volumeObj
+	} else {
+		// If volume object already exists, update the fields such as device path
+		volumeObj.devicePath = devicePath
+		volumeObj.spec = volumeSpec
```

Comment:
I tried to add the unit test, but this device path stored in attachedVolumes does not used anywhere. So right now there is no public API to retrieve this piece of information. If we want to test it, it requires to add a new API which we don't really need right now.